src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -3109,9 +3105,42 @@ public boolean isReflectivePreviewAPI(Element el) {
         PREVIEW
     }
 
+    private boolean isClassOrInterface(Element el) {
+        return el != null && (el.getKind().isClass() || el.getKind().isInterface());
+    }
+
     private boolean hasNoProviewAnnotation(Element el) {
         return el.getAnnotationMirrors()
                  .stream()
                  .anyMatch(am -> "jdk.internal.javac.NoPreview".equals(getQualifiedTypeName(am.getAnnotationType())));
     }
+
+    private PreviewFlagProvider previewFlagProvider = new PreviewFlagProvider() {
+        @Override
+        public boolean isPreview(Element el) {
+            PreviewSummary previewAPIs = declaredUsingPreviewAPIs(el);
+            Element enclosing = el.getEnclosingElement();
+
+            return (!previewLanguageFeaturesUsed(el).isEmpty() ||
+                    configuration.workArounds.isPreviewAPI(el) ||
+                    (!isClassOrInterface(el) && isClassOrInterface(enclosing) &&
+                    configuration.workArounds.isPreviewAPI(enclosing)) ||
+                    !previewAPIs.previewAPI.isEmpty() ||
+                    !previewAPIs.reflectivePreviewAPI.isEmpty() ||
+                    !previewAPIs.declaredUsingPreviewFeature.isEmpty()) &&
+                   !hasNoProviewAnnotation(el);
```

Comment:
Minor grumbles.

1. Normal JDK code style is to put binary operators at the beginning of a one, not the end of a previous line
2. It would help a bit to use indentation to show the `&&` groups
3.last line is inconsistently indented